tests/functional/tier4/test_compile_linux_kernel_with_self_heal.py

```diff
@@ -0,0 +1,222 @@
+        for brick_arequal in arequals:
+            brick_total = brick_arequal.splitlines()[-1].split(':')[-1]
+            self.assertEqual(init_val, brick_total, 'Arequals not matching')
+
+    @staticmethod
+    def _add_dir_path_to_brick_list(brick_list):
+        """Add test_self_heal at the end of brick path"""
+        dir_brick_list = []
+        for brick in brick_list:
+            dir_brick_list.append('{}/{}'.format(brick,
+                                                 'test_self_heal'))
+        return dir_brick_list
+
+    def _check_arequal_checksum_for_the_volume(self):
+        """
+        Check if arequals of mount point and bricks are
+        are the same.
+        """
+        if self.volume_type == "replicated":
+            # Check arequals for "replicated"
+            brick_list = get_all_bricks(self.mnode, self.volname)
+            dir_brick_list = self._add_dir_path_to_brick_list(brick_list)
+
+            # Get arequal before getting bricks offline
+            work_dir = '{}/test_self_heal/'.format(self.mountpoint)
+            ret, arequals = collect_mounts_arequal([self.mounts[0]],
+                                                   path=work_dir)
+            self.assertTrue(ret, 'Failed to get arequal')
+            g.log.info('Getting arequal before getting bricks offline '
+                       'is successful')
+
+            # Get arequal on bricks and compare with mount_point_total
+            self._check_arequal_on_bricks_with_a_specific_arequal(
+                arequals, dir_brick_list)
+
+        # Check arequals for "distributed-replicated"
+        if self.volume_type == "distributed-replicated":
+            # Get the subvolumes
+            subvols_dict = get_subvols(self.mnode, self.volname)
+            num_subvols = len(subvols_dict['volume_subvols'])
+
+            # Get arequals and compare
+            for i in range(0, num_subvols):
+                # Get arequal for first brick
+                brick_list = subvols_dict['volume_subvols'][i]
+                dir_brick_list = self._add_dir_path_to_brick_list(brick_list)
+                ret, arequals = collect_bricks_arequal([dir_brick_list[0]])
+                self.assertTrue(ret, 'Failed to get arequal on first brick')
+
+                # Get arequal for every brick and compare with first brick
+                self._check_arequal_on_bricks_with_a_specific_arequal(
+                    arequals, dir_brick_list)
+
+    def test_compile_linux_kernel_with_self_heal(self):
+        """
+        Test Case:
+        1. Create a volume of any type, start it and mount it.
+        2. Perform linux untar on the mount point and wait for it to complete.
+        3. Bring down brick processes of volume.
+        4. Compile linux kernel on monut point.
+        5. Bring offline bricks back online.
+        6. Wait for heal to complete.
+        7. Check arequal checksum on bricks and make sure there is
+           no data loss.
+
+        Note:
```

Comment:
Mention the approximate execution time of this test script .

Reply:
Ack